### ISSUE TITLE
Cloud error 411

### ISSUE THREAD

**Issue by aferende** [AlexxIT/SonoffLAN#1533]
Hi, I often find several of these 411 errors in the logs and occasionally some 504 errors (the device id changes)..

```
[W] Cloud ERROR: {'error': 411, 'deviceid': 'a4800aad2b', 'apikey': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322', 'sequence': '1732798064352', 'uid': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322'}
```

```
[W] Cloud ERROR: {'error': 504, 'deviceid': 'a4800aad2b', 'apikey': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322', 'sequence': '1732798064352', 'uid': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322'}
```

Now the 504 (bad gateway) is probably due to a timeout error of the ITHEAD/Sonoff servers in Cloud.
I would like to know if the 411 error depends on the SonOffLan integration or if this is also an error of their Cloud (but what type?)
Http 411 is "Length Required" ... but what does it mean?
Is it receiving a Json payload of the wrong length?

Thanks
[home-assistant_sonoff_2024-11-28T18-02-37.106Z.log](https://github.com/user-attachments/files/17951283/home-assistant_sonoff_2024-11-28T18-02-37.106Z.log)

**Comment by AlexxIT** [comment]
Looks like it's undocumented error https://coolkit-technologies.github.io/eWeLink-API/#/en/OAuth2.0?id=v2-interface-error-code
I have no idea what it means.

**Comment by aferende** [comment]
Ty @AlexxIT 
I wanted to try to verify if this error depends on APPID and APPSECRET. Since I registered on the ewelink developer portal and I have my pair, could you tell me how I can test them in the SONOFFLAN integration? Where in the code should I put them?
Thx,
Andrea

**Comment by AlexxIT** [comment]
```yaml
sonoff:
  appid: xxx
  appsecret: xxx
```

**Comment by aferende** [comment]
I receive this error:

![IMG_20241129_153759.jpg](https://github.com/user-attachments/assets/fe76426a-39ce-4f29-af32-068684c4ca64)

**Comment by AlexxIT** [comment]
So your appid can't use this API

**Comment by aferende** [comment]
I registered here to get appid and secretid: https://dev.ewelink.cc.
Is this the correct link?

**Comment by AlexxIT** [comment]
I have no idea. Never tried her this keys.

**Comment by aferende** [comment]
how did you get yours?

**Comment by AlexxIT** [comment]
Ewelink give it personally for SonoffLAN project.

**Comment by aferende** [comment]
Hi @AlexxIT,
is it possible to know what is the request and the json payload that then triggers this error response? if you help me I'll try to write to the support collkit to find out the nature of this 411 error. 
Thanks,
Andrea

**Comment by AlexxIT** [comment]
Try to enable debug logs for integration

**Comment by aferende** [comment]
I tried both like this:
![image](https://github.com/user-attachments/assets/4559cb95-443d-43d9-b234-67b937ed3c3a)

and like this:
![image](https://github.com/user-attachments/assets/e8213ae4-86de-46d2-84b5-fbadd849bdc6)

but when the error occurs I have no further logs.
![image](https://github.com/user-attachments/assets/e8d33197-2cd8-4045-9d01-35dbff162621)

![image](https://github.com/user-attachments/assets/d0a2ba5f-58f1-4c66-82d0-3950c1dfea1a)

Am I enabling debug logs correctly or is there another way?

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/SonoffLAN?tab=readme-ov-file#debug-page

**Comment by aferende** [comment]
Hi @AlexxIT, 
there seems to be the same information also in debug log but the requests are missing which then generate these 504 and 411 errors in response.
![image](https://github.com/user-attachments/assets/c9029d8b-011f-4ee5-ad9d-15c659a2d407)


[logs.zip](https://github.com/user-attachments/files/18005980/logs.zip)

By the way, coolokit support responded to me a little while ago and wants to have an example of a Json request.
![image](https://github.com/user-attachments/assets/f9f71908-f938-426f-bfcb-9c0194e4b6b9)

How can it be done? Thanks as usual for your support

**Comment by aferende** [comment]
Coolkit support claims that it is not their cloud that responds with error 411 (but I don't believe it).
It becomes essential at this point to give them evidence of the request that the integration sends them

![image](https://github.com/user-attachments/assets/daf473a4-7cda-410b-88e3-8aaf2c0a3997)

**Comment by aferende** [comment]
I found this solution:

![image](https://github.com/user-attachments/assets/571100e8-22a2-4f72-89c7-98175ec2c5e1)

In this way I have full payload to send to coolkit support.

`{'action': 'update', 'apikey': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322', 'selfApikey': 'XXXXXXXXXXX', 'deviceid': 'a4800aad2b', 'params': {'switch': 'on'}, 'userAgent': 'app', 'sequence': '1733313770787'}`

I'll keep you updated
Ty,
Andrea.

**Comment by AlexxIT** [comment]
Are you sure this is request for your errors? Is field `sequence` same?

**Comment by aferende** [comment]
Yes I am sure.
See here.
Can you think of any solutions or possible causes?
Anyone else get this error?

![image](https://github.com/user-attachments/assets/6a06a63c-e30e-4602-a98b-a86992e490b2)

**Comment by AlexxIT** [comment]
But you showing response without error.

**Comment by aferende** [comment]
Sorry I had taken a sequence without errors.
See this one from a few minutes ago:

![image](https://github.com/user-attachments/assets/108f75e9-7b91-439a-904e-32a16acc84e8)

**Comment by AlexxIT** [comment]
It's bad you're wiping out device IDs. Was the first line (local query) sent with the same identifier? It ended with an error, too.

**Comment by aferende** [comment]
Sorry. It's just that I still haven't figured out if the device id is sensitive data or not :-)
I don't think so, since it's in the debug info.
In any case, this is the text that is visible in plain text:

```
2024-12-04 15:14:44 [D] 10022bd4a5 => Local4 | 192.168.100.223:8081 | {'switch': 'off', 'subDevId': 'a4800aad2b'} <= {'sequence': '1733321684291', 'seq': 2, 'error': 400, 'encrypt': True}
2024-12-04 15:14:44 [D] a4800aad2b => Cloud4 | {'switch': 'off'} | 1733321684291
2024-12-04 15:14:44 [D] a4800aad2b => Cloud4 | {'switch': 'off'} | 1733321684291 payload: {'action': 'update', 'apikey': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322', 'selfApikey': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322', 'deviceid': 'a4800aad2b', 'params': {'switch': 'off'}, 'userAgent': 'app', 'sequence': '1733321684291'} 
2024-12-04 15:14:47 [W] Cloud ERROR: {'error': 411, 'deviceid': 'a4800aad2b', 'apikey': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322', 'sequence': '1733321684291', 'uid': '34d8fdc6-4a0d-4771-a2d2-ee0d9a5b5322'}
2024-12-04 15:14:47 [D] 10022bd4a5 => Local4 | 192.168.100.223:8081 | None <= {'sequence': '1733321687578', 'seq': 2, 'error': 400, 'encrypt': True}

```

In fact, I'm attaching the latest complete debug where there are other 411 errors:
[logs.zip](https://github.com/user-attachments/files/18011791/logs.zip)

Let me know what you think

**Comment by aferende** [comment]
Is it possibile we are exceeding the limit of 50,000 calls per month?
Only in the log I attached in the previous message there are 215 calls in 5 hours.
Doing a quick count, I alone make 30,000 calls per month (I have 27 sonoff devices).
In the coolkit documentation they say that for each cloud site you can make (free) a maximum of 50,000 calls per month.

![image](https://github.com/user-attachments/assets/bf5ff65d-e810-4024-b83f-016f58c281a7)

Are these 50,000 calls per account or overall per Appid?

**Comment by AlexxIT** [comment]
Don't think so. My devices works fine over cloud.

You have problems ONLY with ZBMINIL2. Looks like you have Zigbee Pro bridge.
Do you never have these devices working at all? Or do they only have certain commands that don't work?

**Comment by aferende** [comment]
I confirm that I have a Zigbee Pro Bridge (with a recently updated signature to version 2.2.0 - see bug  https://github.com/AlexxIT/SonoffLAN/issues/1529)
These devices work with all commands but sometimes (several times a day) they give these errors (411,504).
In addition to the error in the log, obviously the effect is that the light does not turn on or a sensor remains stuck on "motion detected" and does not update its state in cloud.
The only way to solve it is to turn the light on/off again or pass near the sensor (after 1 minute) to force the status update in the cloud.
The only peculiarity is that in all ZGMINIL2 I have activated the RSSI entities (which are disabled by default) just to monitor signal quality.

**Comment by aferende** [comment]
Hi @AlexxIT, this is the answer given by ewelink support.
What do you think about?

![image](https://github.com/user-attachments/assets/485ee861-ee0b-4f69-8a13-872e5182b47f)

**Comment by AlexxIT** [comment]
There's a logic to it.  Signal strength isn't always the most important thing. Even with a strong signal, commands may not be received. It's radio. Many devices can cause interference.

**Comment by aferende** [comment]
**It's really incredible. I managed to find the cause of the 411 error and solve the problem.**
**It was actually an interference problem on Zigbee channel 25.**

I should point out that I have a SonOff Bridge Pro that in its settings allows you to change the Zigbee channel and is also able to scan the best channel.
The scan always suggested that channel 25 was the most free from interference and therefore it was on that channel.
In reality, the problem was exactly this; on this channel I had a lot of interference probably due to the many wifi access points that reach me from the neighborhood.

![image](https://github.com/user-attachments/assets/42b8552b-ac6b-4dfe-b55d-22f18a874fd7)

I had already configured my WiFi on a channel whose interference did not overlap with that of the Zigbee.
In short, after several 411 errors that were driving me crazy, I decided to change the Zigbee channel again.
Every time you change the Zigbee channel, all the devices are automatically removed and reset and must be re-added manually.
During this process, I did a fundamental test.

**I tried all the channels one by one and tried to add all the devices together (for each single channel).**

From here I realized that the only channel with which all the devices were added was channel 15, which the pro bridge recommended because it was indicated as the most crowded.

After changing the channel now I finally have all the devices stable and for 3 days not even a 411 error (before I had several a day).

Now incredibly the Pro Bridge reports channel 25 as the most crowded. Something must not be working well in this algorithm.

![image](https://github.com/user-attachments/assets/7bcd8a1a-8747-44f0-942a-1ca5cc62bda1)

Thanks to the ewelink support that reported to me that the 411 error (not documented) concerns unreachable devices and to @AlexxIT support now everything is stable.
I hope that this information can help others who have encountered this problem.

I also leave the link with the guide on how to avoid interference between zigbee and wifi.

**ZigBee Network Optimization (and Repeaters/Router) Mesh**: https://community.home-assistant.io/t/zigbee-network-optimization-a-how-to-guide-for-avoiding-radio-frequency-interference-adding-zigbee-router-devices-repeaters-extenders-to-get-a-stable-zigbee-network-mesh-with-best-possible-range-and-coverage-by-fully-utilizing-zigbee-mesh-networking/515752

![image](https://github.com/user-attachments/assets/1999387d-9bba-48d6-b9e2-297b96c9f0e9)

I think this thread can be closed.
Thx.
